src/azure-client.ts

```diff
@@ -0,0 +1,48 @@
+import * as fs from "fs";
+
+import FormData from "form-data";
+
+import { getApiClient } from "./gh-api-client";
+
+export interface Policy {
+  upload_url: string;
+  header: Map<string, string>;
+  form: Map<string, string>;
+}
+
+export async function uploadArtifact(policy: Policy, artifactZipPath: string) {
+  const data = new FormData();
+  for (const [key, value] of policy.form.entries()) {
+    data.append(key, value);
+  }
+
+  const artifactContents = fs.readFileSync(artifactZipPath, "utf8");
+  data.append("file", artifactContents, {
+    contentType: "application/zip",
+    filename: "results.zip",
+  });
+
+  const httpClient = getApiClient();
+
+  const additionalHeaders = {
+    ...policy.header,
+    ...data.getHeaders(),
+  };
+
+  const response = await httpClient.sendStream(
+    "POST",
+    policy.upload_url,
+    data,
+    additionalHeaders
+  );
+
+  if (!response.message.statusCode || response.message.statusCode > 299) {
```

Comment:
Just a note to other readers that this response is coming from azure / memory-alpha, so we aren't as directly in control of it. I don't currently know exactly what status code we get here, but I don't think it matters too much to specify exactly what we expect here. Anything in the 2xx range should indicate success. So I support just leaving this code as it is.